packages/ui/core-components/src/lib/unsorted/viz/map/components/Legend Components/LegendIcons.svelte

```diff
@@ -0,0 +1,40 @@
+<script>
+	/** @type {boolean} */
+	export let hideLegend = false;
+	export let handleLegendClick;
+
+	/** @type {string | undefined} */
+	export let legendType = undefined;
+	/** @type {'left' | 'right' | 'bottom' | 'top'} */
+	export let direction = 'bottom';
+
+	const chevronDirections = {
+		top: { show: 'rotate-[270deg]', hide: 'rotate-90' },
+		bottom: { show: 'rotate-90', hide: 'rotate-[270deg]' },
+		left: { show: 'rotate-0', hide: 'rotate-180' },
+		right: { show: 'rotate-180', hide: 'rotate-0' }
+	};
+</script>
+
+<button
+	class="items-center {legendType === 'scalar' ? 'h-10 px-[2.9px]' : `h-5 pl-2 pt-0.5 w-full`}"
+	on:click={handleLegendClick}
+	on:dblclick={(e) => e.stopPropagation()}
+>
```

Comment:
Add an `aria-label` to this button that says either `Expand Legend` or `Collapse Legend` so that screenreader users know what it does